compiler/rustc_hir_analysis/src/lib.rs

```diff
@@ -494,6 +494,18 @@ pub fn check_crate(tcx: TyCtxt<'_>) -> Result<(), ErrorGuaranteed> {
         tcx.hir().for_each_module(|module| tcx.ensure().check_mod_item_types(module))
     });
 
+    // FIXME: Parallelize other iterators(e.g. `body_owners` in `check_unused`) instead
```

Comment:
Actually it doesn't change the logic, because `check_unused` will call all `typeck` and fetch the `used_trait_impl` information in the results. I moved this part into `check_unused` to make it look more reasonable.

Reply:
I just add a FIXME above `check_unused` now